From 5883ee8afd65e46c658031ef25d5de5ecfe5361a Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Tue, 30 Dec 2025 18:18:06 +0800
Subject: [PATCH 49/63] loonggpu-bridge: remove calls to drm_do_get_edid

It's removed in 6.10 kernel release.

I don't know if the change is correct as I've no device to test it.

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 loonggpu-bridge/it66121_drv.c | 12 ++++--------
 loonggpu-bridge/lt8618_drv.c  |  9 ++-------
 2 files changed, 6 insertions(+), 15 deletions(-)

diff --git a/loonggpu-bridge/it66121_drv.c b/loonggpu-bridge/it66121_drv.c
index bef45bb..8062dd2 100644
--- a/loonggpu-bridge/it66121_drv.c
+++ b/loonggpu-bridge/it66121_drv.c
@@ -351,17 +351,13 @@ static int it66121_get_edid_block(void *data, u8 *buf, unsigned int block,
 static int it66121_get_modes(struct loonggpu_bridge_phy *phy,
 			     struct drm_connector *connector)
 {
-	struct edid *edid;
-	unsigned int count = 0;
-	edid = drm_do_get_edid(connector, it66121_get_edid_block, phy);
-	if (edid) {
-		drm_connector_update_edid_property(connector, edid);
-		count = drm_add_edid_modes(connector, edid);
-	} else {
+	unsigned int count;
+
+	count = drm_edid_connector_add_modes(connector);
+	if (!count) {
 		count = drm_add_modes_noedid(connector, 1920, 1080);
 		DRM_DEV_INFO(to_dev(phy), "Update default edid method 1080p\n");
 	}
-	kfree(edid);
 	return count;
 }
 
diff --git a/loonggpu-bridge/lt8618_drv.c b/loonggpu-bridge/lt8618_drv.c
index cfe043e..d3d18fd 100644
--- a/loonggpu-bridge/lt8618_drv.c
+++ b/loonggpu-bridge/lt8618_drv.c
@@ -624,15 +624,10 @@ static int lt8618_get_edid_block(void *data, u8 *buf, unsigned int block,
 static int lt8618_get_modes(struct loonggpu_bridge_phy *phy,
 			    struct drm_connector *connector)
 {
-	struct edid *edid;
 	unsigned int count;
 
-	edid = drm_do_get_edid(connector, lt8618_get_edid_block, phy);
-	if (edid) {
-		drm_connector_update_edid_property(connector, edid);
-		count = drm_add_edid_modes(connector, edid);
-		kfree(edid);
-	} else {
+	count = drm_edid_connector_add_modes(connector);
+	if (!count) {
 		count = drm_add_modes_noedid(connector, 1920, 1080);
 		drm_set_preferred_mode(connector, 1024, 768);
 		DRM_DEBUG("Update default edid method\n");
-- 
2.52.0

